File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.swift
import Foundation

func solution(_ s:String) -> Int {
    
    if s.count <= 2 { return 1 }
    
    var result = 0, same = 0, diff = 0
    var arr = Array(s)
    var first = arr[0]
    
    for element in arr {
        if same == diff { first = element }
        (first == element) ? (same += 1) : (diff += 1)
        if same == diff { result += 1 }
    }
    
    if same != diff { result += 1 }
    
    return result
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0, same = 0, diff = 0;
    string word = "";
    
    for (char ch : s) {
        string str = to_string(ch);
        if (word.empty()) {
            word = str;
        }

        (word == str) ? same += 1 : diff += 1;

        if (same == diff) {
            answer += 1;
            word = "";
            same = 0;
            diff = 0;
        }
    }

    if (same != 0 || diff != 0) {
        answer += 1;
    }

    return answer;
}